### DISCUSSION TITLE
Deserialization to class not working

### DISCUSSION THREAD

**Discussion by rpresser** [reactiveui/refit#1728] [Q&A] [answered]
I am trying to capture the [Digicert Automation API](https://dev.digicert.com/en/certcentral-apis/automation-api.html) with a Refit 
HttpClient Factory. 

When I define the method to return string, so I can see the raw JSON returned, it appears correct
`Task<string> GetAgentList([Body] GetAgentListReq req);`
Returns JSON string that starts off like this:
```
{"error":null,"data":{"agentList":[{"agentId":2545.0,"accountId":"1349264","divisionId":"91804","hostname":"MTLWPVINSIWEB12","agentName":"10.220.100.37","status":"Configured","sensorLicenseKey":"39F3356652833A47","sensorName":"mtlwdvinsidemo_20200921","hostIp":"10.220.100.37","type":"Agent","agentLicense":"83C7BF7AED8F7358","updatePreference":0.0,"updateAvailable":false,"tag":"","osName":"Windows"},
```
But when I change it to `Task<AgentList>`, the result deserializes without error but the `data` value is null. My interface file and my console client app are attached.
[IDigicert.Automation.API.cs.zip](https://github.com/user-attachments/files/15928385/IDigicert.Automation.API.cs.zip)
[Program.zip](https://github.com/user-attachments/files/15928394/Program.zip)

**Comment by glennawatson** [comment, marked as answer]
A null inner property after a successful (no-error) deserialize almost always means a name mismatch between the JSON and your model, not a Refit bug.

In your payload the shape is:

```json
{ "error": null, "data": { "agentList": [ ... ] } }
```

By default Refit uses System.Text.Json, which is case-sensitive and does not map snake_case automatically. So a property named Data will not bind to "data" unless you configure it.

Fixes (pick one):

- Make matching case-insensitive (and snake_case if needed) by passing JsonSerializerOptions when you build the client:

```csharp
var options = new JsonSerializerOptions
{
    PropertyNameCaseInsensitive = true,
    PropertyNamingPolicy = JsonNamingPolicy.CamelCase
};

var api = RestService.For<IDigicert>(httpClient, new RefitSettings
{
    ContentSerializer = new SystemTextJsonContentSerializer(options)
});
```

- Or annotate each property explicitly:

```csharp
public class AgentListResponse
{
    [JsonPropertyName("error")] public string? Error { get; set; }
    [JsonPropertyName("data")] public AgentListData? Data { get; set; }
}
```

Also note agentId comes back as 2545.0 and updatePreference as 0.0 (floating point), so model those as double, not int, or deserialization of that node can fail/return defaults.

If after matching the names exactly (or enabling case-insensitive) data is still null, post your model classes inline and we can pinpoint it.
